lib/driver.js

```diff
@@ -281,10 +292,35 @@ class AndroidUiautomator2Driver extends BaseDriver {
     if (!this.opts.systemPort) {
       return;
     }
-    await PORT_ALLOCATION_GUARD.acquire(AndroidUiautomator2Driver.name,
+    await DEVICE_PORT_ALLOCATION_GUARD.acquire(AndroidUiautomator2Driver.name,
       async () => await this.adb.removePortForward(this.opts.systemPort));
   }
 
+  async allocateMjpegServerPort () {
```

Comment:
I don't see a reason to always automatically allocate this port if `this.opts.mjpegServerPort` is not provided explicitly. the automatically allocated port number number does not make much sense to an external client, since it is internal to the driver

Reply:
Makes sense. I have refactored how this bit works, please re-review w/ the latest commit.